pyam/core.py

```diff
@@ -405,6 +405,28 @@ def validate(self, criteria={}, exclude=False):
 
             return df
 
+    def conv_unit(self, unit_conv, inplace=False):
+        """ Converts units based on provided unit conversion factors
+
+        Parameters
+        ----------
+        unit_conv: dict
+            for each unit for which a conversion should be carried out, provide current unit and target unit and conversion factor
+            {<current unit>: [<target unit>, <conversion factor>]}
+        inplace: bool, default False
+            if True, do operation inplace and return None
+        """
+        ret = copy.deepcopy(self) if not inplace else self
+        for unt in unit_conv:
+            if not type(unit_conv[unt][1]) in [int, np.integer, float]:
```

Comment:
here just use pd.as_numeric(factor) and let the error pass through to user if there is one

Reply:
i.e., 

```
factor = pd.as_numeric(factor)
```